### DISCUSSION TITLE
Question about importing with custom meta fields

### DISCUSSION THREAD

**Discussion by nicjp** [kimai/kimai#2600] [General]
Hello;

I am using Kimai version 1.4.3 with the "Custom Fields" plugin and am attempting to use the command-line import tool to import a list of projects. 

On import, all data is imported as expected except the data for the Custom Fields. 

For example, I have a custom field whose "internal name" is "project_code". In my CSV file, I have a column called: meta.project_code

On import, this data is not present, and in the database, a record has been created but with a NULL value. 

Would appreciate any advice on how to make this work!

Cheers,

Nic

**Comment by kevinpapst** [comment]
Hey Nic, generally: the plugin is not required to import meta fields. 
The code looks ok to me. Please share a CSV file.

**Reply by nicjp** [reply]
Hey Kevin, 

Thanks for the info. 

I've attached an example of the CSV that I'm using. 

[project-import.csv](https://github.com/kevinpapst/kimai2/files/6616050/project-import.csv)

**Reply by kevinpapst** [reply]
Works as expected with `bin/console kimai:import:project ~/Downloads/project-import.csv`:
<img width="603" alt="Bildschirmfoto 2021-06-08 um 13 33 40" src="https://user-images.githubusercontent.com/533162/121177815-24835580-c85e-11eb-9662-5fd82ae33abb.png">

<img width="651" alt="Bildschirmfoto 2021-06-08 um 13 33 30" src="https://user-images.githubusercontent.com/533162/121177789-1f260b00-c85e-11eb-98c7-68f59deec263.png">

**Reply by nicjp** [reply]
Hi Kevin;

That's the command I'm using - however following an import I get NULL values in the VALUE column of the kimai2_projects_meta table:

```
MariaDB [kimai]> select * from kimai2_projects_meta
    -> ;
+----+------------+--------------+-------+---------+
| id | project_id | name         | value | visible |
+----+------------+--------------+-------+---------+
| 27 |         17 | project_code | 3013A |       1 |
| 30 |         47 | project_code | NULL  |       1 |
| 31 |         48 | project_code | NULL  |       1 |
+----+------------+--------------+-------+---------+
```

Cheers,

Nic

**Reply by kevinpapst** [reply]
As long as I cannot reproduce the problem, I am not able to help.
You need to provide more information.
Delete the rows in the kimai2_projects_meta table and re-run the command. What happens and what is the commands output?

**Reply by nicjp** [reply]
Hi Kevin;

Hopefully this will help - I did what you asked and experienced the same behaviour. 

I then deleted the field definition created using the CustomMeta plugin, and tried again, and the values load as expected. 

I'm unsure what the correlation is however, but that is the behaviour I am currently seeing. 

Cheers,

Nic

**Reply by kevinpapst** [reply]
Hm, okay. I haven't tried that before. I'll investigate, thanks for your feedback!
Glad that it works now 👍
